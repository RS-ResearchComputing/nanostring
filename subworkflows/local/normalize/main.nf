//
// Perform normalizations (currently using Nacho)
//

include { NACHO_NORMALIZE } from '../../../modules/local/nacho/normalize'

workflow NORMALIZE {
    take:
    counts      // channel: path(rcc files)
    samplesheet // file: /path/to/samplesheet.csv

    main:
    NACHO_NORMALIZE ( counts, samplesheet )

    emit:
    normalized_counts       = NACHO_NORMALIZE.out.normalized_counts        // channel: [ normalized_counts.tsv ]
    normalized_counts_wo_HK = NACHO_NORMALIZE.out.normalized_counts_wo_HK  // channel: [ normalized_counts_wo_HK.tsv ]
    versions                = NACHO_NORMALIZE.out.versions                 // channel: [ versions.yml ]
}
