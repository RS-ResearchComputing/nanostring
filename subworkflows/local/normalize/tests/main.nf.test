nextflow_workflow {

    name "Test Subworkflow NORMALIZE"
    script "../main.nf"
    workflow "NORMALIZE"

    test("test profile") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_01_01.RCC', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_02_02.RCC', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_03_03.RCC', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'nanostring/RCC_files/salmon_04_04.RCC', checkIfExists: true)
                    ]
                )
                input[1] = file(params.pipelines_testdata_base_path + 'nanostring/samplesheets/samplesheet_test.csv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // workflow.out.nacho_qc_multiqc_metrics[0].collect { file(it).name }, //undeterministic .png and .txt mqc files
                    workflow.out.normalized_counts,
                    workflow.out.normalized_counts_wo_HK,
                    workflow.out.versions
                    ).match()
                }
            )
        }
    }
}
