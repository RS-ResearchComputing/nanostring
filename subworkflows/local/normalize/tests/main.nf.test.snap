{
    "test profile": {
        "content": [
            [
                "normalized_counts.tsv:md5,a0124c7a24bd04296f441d9ade82a05f"
            ],
            [
                "normalized_counts_wo_HKnorm.tsv:md5,5a2ce112c24e1b0d0f4cf3392111ef9e"
            ],
            [
                "versions.yml:md5,eacb31511724a946730536d6bb82ae6d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T23:22:51.717778934"
    }
}